Test/test codebase

guor8lei opened the pull request:
Fixes #1 

Establishes a small test Angular application with some anti-patterns for each of the four flags that this tool will attempt to programatically fix. More unit test cases will need to be added, but this establishes a good starting point to experiment with the TS compiler API and ts-morph.

jiady commented on test/test-app/src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,27 @@
+
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
```
remove extra line

jiady commented on test/test-app/src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,27 @@
+
+
+const routes: Routes = [];
+
```
add docs for all "export"

jiady commented on test/test-app/src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,27 @@
+/*
```
let's name x_x_x.ts

app_routing_module.ts

if Module is for a specific component, we don't need an extra module file.

jiady commented on test/test-app/src/app/app.component.css:
```diff
@@ -0,0 +1,15 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
```
let's add a simple style or remove the file.

guor8lei replied:
I'll remove all CSS files for now, since styling shouldn't really be a factor in testing this tool.

jiady commented on test/test-app/src/app/app.component.spec.ts:
```diff
@@ -0,0 +1,51 @@
+/*
```
app_component_test.ts

I don't have any objection for writing test, but I don't want to you focus too much on it. Since this is already an app for test, if you write test, you should violate the compiler rules in test.

guor8lei replied:
I agree, I'll delete all the other test files for now and just leave app_component_test.ts as the only test file, which I put some anti-patterns in as well.

jiady commented on test/test-app/src/app/app.module.ts:
```diff
@@ -0,0 +1,42 @@
+/*
```
put this to app_component.ts

jiady commented on test/test-app/src/app/app.component.ts:
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
```
app_component.ts

guor8lei replied:
@jiady Is there a reason why we want to use underscores for file names over hyphens? I thought that convention was to do `app-component.ts`

jiady commented on test/test-app/src/app/test-no-implicit-any/test-no-implicit-any.component.css:
```diff
@@ -0,0 +1,15 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
```
remove or add simple style.

jiady commented on test/test-app/src/app/test-no-implicit-any/test-no-implicit-any.component.ts:
```diff
@@ -0,0 +1,98 @@
+    this.calledByPassString(val);
+  }
+
+  calledByPassString(val) {}
```
add comments below or above on what it should be, so is others.

jiady commented on test/test-app/src/app/test-no-implicit-returns/test-no-implicit-returns.component.css:
```diff
@@ -0,0 +1,15 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
```
remove or add simple style

jiady commented on test/test-app/src/app/test-no-implicit-returns/test-no-implicit-returns.component.html:
```diff
@@ -0,0 +1,22 @@
+    limitations under the License.
+-->
+
+<p>--noImplicitReturns</p>
```
h6 instead of p?

jiady commented on test/test-app/src/app/test-strict-null-checks/test-strict-null-checks.component.ts:
```diff
@@ -0,0 +1,104 @@
+})
+export class TestStrictNullChecksComponent implements OnInit {
+  constructor() {}
+
```
let's add a class member variable(which does not have type undefined.) and assign it to null/undefined

jiady commented on test/test-app/src/app/test-strict-null-checks/test-strict-null-checks.component.ts:
```diff
@@ -0,0 +1,104 @@
+
+  passesNullValue() {
+    this.doesNotExpectNull(null);
+  }
```
let's add another one

passesNullVaiable() {
    let n: number|null = null.
    this.doesNotExpectNull(n);
  }


passesUndefinedVaiable() {
    let n: number|undefined = undefined.
    this.doesNotExpectNull(n);
  }

jiady commented on test/test-app/src/app/test-strict-null-checks/test-strict-null-checks.component.ts:
```diff
@@ -0,0 +1,104 @@
+  // Test: Add element to empty list
+  addToEmptyList() {
+    const emptyList = [];
+    emptyList.push(5);
```
do you mean push(null)?

guor8lei replied:
The purpose of this test was to add a number (or any primitive value) to an empty list. The fix would be to explicitly make the type of `emptyList` to be `number[]`. I made a comment about it in the code

jiady commented on test/test-app/src/app/test-strict-property-initialization/test-strict-property-initialization.component.ts:
```diff
@@ -0,0 +1,42 @@
+  ngOnInit(): void {}
+}
+
+interface TestInterface {
```
let's define this in a separate file, may be common/interface

I believe you will need it in other case as well.

guor8lei commented:
@jiady Finish addressing all of the comments and suggestions, I had a question regarding using underscores or hyphens for file-naming, but other than that, let me know if there is anything else!
